frontend/app/src/App.tsx

```diff
@@ -1303,12 +1300,20 @@ export class App extends PureComponent<Props, State> {
 
     const { currentPageScriptHash } = this.state
     const { basePath } = baseUriParts
-    const queryString = this.getQueryString()
+    let queryString = this.getQueryString()
     let pageName = ""
 
     if (pageScriptHash) {
       // The user specified exactly which page to run. We can simply use this
       // value in the BackMsg we send to the server.
+      if (pageScriptHash != currentPageScriptHash) {
+        // reset query string when changing pages to replicate real web apps
```

Comment:
I think this comment might as well be removed. It's already clear what's going on here, and the comment isn't particularly relevant to what's actually happening here since this isn't even where we're doing the query string resetting. That happens above, and this is just where we let the host know